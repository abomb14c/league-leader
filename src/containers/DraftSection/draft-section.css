.draft-container {
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 1200px;
  width: 100%;
}

.draft-section {
  display: flex;
  padding-left: 5%;
  width: 100%;
}

.draft-teams {
  box-shadow: 0px 2px 10px 0px rgba(0, 0, 0, 0.30);
  display: flex;
  flex-direction: column;
  height: 905px;
  margin-left: 5%;
  overflow: scroll;
  width: 17%;
}

.draft-teams-title {
  align-items: center;
  border-bottom: 2px solid #C63D0F;
  background-color: #3B3738;
  display: flex;
  height: 42px;
  padding-left: 2%;
  padding-bottom: 10%;
  padding-top: 10%;
}

.league-info-container {
  align-items: center;
  background-color: #3B3738;
  box-shadow: 0px 2px 10px 0px rgba(0, 0, 0, 0.30);
  display: flex;
  flex-direction: column;
  height: 150px;
  margin-bottom: 2%;
  width: 25%;
 
}

.title-desc {
  color: #FDF3E7;
}

.league-bet {
  color: #7E8F7C;
  font-weight: 200;
  margin-top: 3%;
}

.league-bet-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 100%;
}

.league-bet-desc {
  color: #FDF3E7;
  font-weight: 200;
  margin: 0;
}

.league-title {
  color: #FDF3E7;
}