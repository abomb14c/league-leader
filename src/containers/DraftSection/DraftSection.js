import React, { Component } from 'react';
import { connect } from 'react-redux';

import './draft-section.css';
import UserDraftCard from '../UserDraftCard/UserDraftCard';
import DraftTeamsContainer from '../DraftTeamsContainer/DraftTeamsContainer';
import PropTypes from 'prop-types';

export class DraftSection extends Component {
  constructor(props) {
    super(props);
  }
 

  render() {
    return (
      <div className="draft-container">
        <div className="league-info-container">
          <h1 className="league-title">{this.props.league.name}</h1>
          <div className="league-bet-container">
            <h3 className="league-bet-desc">Wager:</h3>
            <h3 className="league-bet">{this.props.league.bet}</h3>
          </div>
        </div>
        <button className="submit-league-button">Submit League</button>
        <div className="draft-section">
          <div className="draft-teams">
            <div className="draft-teams-title">
              <h3 className="title-desc">Available Teams</h3>
            </div>
            <DraftTeamsContainer /> 
          </div>
          <UserDraftCard />
        </div>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  league: state.league,
  draftTeams: state.draftTeams
});

export default connect(mapStateToProps, null)(DraftSection);

DraftSection.propTypes = {
  league: PropTypes.object,
  draftTeams: PropTypes.arrayOf(PropTypes.object)
};