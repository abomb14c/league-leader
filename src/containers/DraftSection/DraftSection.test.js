import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import  { DraftSection, mapStateToProps}  from './DraftSection';

describe('DraftCard', () => {
  let wrapper;
  let mockProps;
  let mockHandleLeagueID;
  beforeEach(() => {
    mockHandleLeagueID = jest.fn();
    mockProps = {
      league: {},
      draftTeams: []
    };

    wrapper = shallow(<DraftSection
      handleLeagueID={mockHandleLeagueID}
      {...mockProps}
    />);
  });
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
  it('should map the user to props', () => {
    const mockState = {
      league: {},
      draftPicks: [{}, {}, {}],
      text: '',
      user: {}
    };

    const expected = {
      league: {},
      draftPicks: [{}, {}, {}],
      user: {}
    };
    
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});
