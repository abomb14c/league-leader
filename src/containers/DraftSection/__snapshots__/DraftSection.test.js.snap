// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DraftCard should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DraftSection
    draftTeams={Array []}
    handleLeagueID={[Function]}
    league={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="league-info-container"
>
          <h1
                    className="league-title"
          />
          <div
                    className="league-bet-container"
          >
                    <h3
                              className="league-bet-desc"
                    >
                              Wager:
                    </h3>
                    <h3
                              className="league-bet"
                    />
          </div>
</div>,
        <button
          className="submit-league-button"
          onClick={[Function]}
>
          Submit League
</button>,
        <div
          className="draft-section"
>
          <div
                    className="draft-teams"
          >
                    <div
                              className="draft-teams-title"
                    >
                              <h3
                                        className="title-desc"
                              >
                                        Available Teams
                              </h3>
                    </div>
                    <Connect(Component) />
          </div>
          <Connect(UserDraftCard) />
</div>,
      ],
      "className": "draft-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="league-title"
/>,
            <div
              className="league-bet-container"
>
              <h3
                            className="league-bet-desc"
              >
                            Wager:
              </h3>
              <h3
                            className="league-bet"
              />
</div>,
          ],
          "className": "league-info-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "league-title",
            },
            "ref": null,
            "rendered": null,
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3
                  className="league-bet-desc"
>
                  Wager:
</h3>,
                <h3
                  className="league-bet"
/>,
              ],
              "className": "league-bet-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Wager:",
                  "className": "league-bet-desc",
                },
                "ref": null,
                "rendered": "Wager:",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "league-bet",
                },
                "ref": null,
                "rendered": null,
                "type": "h3",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit League",
          "className": "submit-league-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Submit League",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="draft-teams"
>
              <div
                            className="draft-teams-title"
              >
                            <h3
                                          className="title-desc"
                            >
                                          Available Teams
                            </h3>
              </div>
              <Connect(Component) />
</div>,
            <Connect(UserDraftCard) />,
          ],
          "className": "draft-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="draft-teams-title"
>
                  <h3
                                    className="title-desc"
                  >
                                    Available Teams
                  </h3>
</div>,
                <Connect(Component) />,
              ],
              "className": "draft-teams",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h3
                    className="title-desc"
>
                    Available Teams
</h3>,
                  "className": "draft-teams-title",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Available Teams",
                    "className": "title-desc",
                  },
                  "ref": null,
                  "rendered": "Available Teams",
                  "type": "h3",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="league-info-container"
>
            <h1
                        className="league-title"
            />
            <div
                        className="league-bet-container"
            >
                        <h3
                                    className="league-bet-desc"
                        >
                                    Wager:
                        </h3>
                        <h3
                                    className="league-bet"
                        />
            </div>
</div>,
          <button
            className="submit-league-button"
            onClick={[Function]}
>
            Submit League
</button>,
          <div
            className="draft-section"
>
            <div
                        className="draft-teams"
            >
                        <div
                                    className="draft-teams-title"
                        >
                                    <h3
                                                className="title-desc"
                                    >
                                                Available Teams
                                    </h3>
                        </div>
                        <Connect(Component) />
            </div>
            <Connect(UserDraftCard) />
</div>,
        ],
        "className": "draft-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="league-title"
/>,
              <div
                className="league-bet-container"
>
                <h3
                                className="league-bet-desc"
                >
                                Wager:
                </h3>
                <h3
                                className="league-bet"
                />
</div>,
            ],
            "className": "league-info-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "league-title",
              },
              "ref": null,
              "rendered": null,
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3
                    className="league-bet-desc"
>
                    Wager:
</h3>,
                  <h3
                    className="league-bet"
/>,
                ],
                "className": "league-bet-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Wager:",
                    "className": "league-bet-desc",
                  },
                  "ref": null,
                  "rendered": "Wager:",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "league-bet",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h3",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit League",
            "className": "submit-league-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Submit League",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="draft-teams"
>
                <div
                                className="draft-teams-title"
                >
                                <h3
                                                className="title-desc"
                                >
                                                Available Teams
                                </h3>
                </div>
                <Connect(Component) />
</div>,
              <Connect(UserDraftCard) />,
            ],
            "className": "draft-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="draft-teams-title"
>
                    <h3
                                        className="title-desc"
                    >
                                        Available Teams
                    </h3>
</div>,
                  <Connect(Component) />,
                ],
                "className": "draft-teams",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <h3
                      className="title-desc"
>
                      Available Teams
</h3>,
                    "className": "draft-teams-title",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Available Teams",
                      "className": "title-desc",
                    },
                    "ref": null,
                    "rendered": "Available Teams",
                    "type": "h3",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
